Sources/Testing/Issues/Issue.swift

```diff
@@ -93,9 +94,9 @@ public struct Issue: Sendable {
   ///     occurred. This defaults to a default source context returned by
   ///     calling ``SourceContext/init(backtrace:sourceLocation:)`` with zero
   ///     arguments.
-  init(
+  public init(
```

Comment:
This is fine but will produce a source context for the issue that points into Swift Testing rather than the call site, which is an oversight of the existing code to be fair. How about we expose a different initializer with the signature:

```swift
init(
  _ kind: Kind,
  comments: [Comment] = [],
  sourceLocation: SourceLocation = #_sourceLocation
)
```

(Whether or not `kind` deserves a label is something we can bikeshed.)